### PR TITLE
docs: rework documentation

### PR DESCRIPTION
**Pull request by legau**

### CODE REVIEW

File: docs/features/commands.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+and
+
+- ``manage.py grpcrunaioserver``
+
+This command is similar to django's ``manage.py runserver``, except it launches a grpcaioserver in async mode.
+it accepts the following arguments:
+
+- ``address`` : Optional address for which to open a port.
+- ``--max-workers``: Number of maximum worker threads.
```
[the remark this answers is not in the document]
Comment (AMontagu) [reply docs/features/commands.rst:28]: this options should be disable then ? I do it.

File: docs/features/commands.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- ``--dev`` Run the server in development mode. This tells Django to use the auto-reloader and run checks.
+
+
+finally:
+
+- ``manage.py grpcrunserver``
+
+Same as ``grpcrunaioserver`` except this one is for sync mode
```
[the remark this answers is not in the document]
Comment (AMontagu) [reply docs/features/commands.rst]: need more information. Not really sure it need to be regrouped.

File: docs/features/filters.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
+Filters
+==========
+
+Description
+-----------
+
+This page will explain how to set up filters in your app services.
```
[the remark this answers is not in the document]
Comment (AMontagu) [reply docs/features/filters.rst]: almost completely rewrite it.
To read again

**Comment by AMontagu** [comment]
@legau @markdoerr current state of doc:
- Make modifications for almost all legau configuration
- Improved filters page
- Fix all the warnings of the sphinx static file generation
- Added docker for having doc in local (with auto build) to help debug and tests
- merged the how to make custom retrieve
- Rewrite excpetions doc with new changes
- Update django doc link to v5.0

What still missing for merging this:
- Pass over all the :TODO: let by @markdoerr 
- finish the last unresolved from @legau 

Will try to do before the end of the week.

**Comment by markdoerr** [comment]
Wonderful, @AMontagu,
the documentation is shaping and am glad, that you revealed many of the hidden "treasures" of DSG :)
Regarding the TODOs : at these points, experience / expertise is missing from my side, so I hope that one of you can explain these concepts better ( what have you meant by "pass over all the :TODO:" .... ? - just "going through" ?  ).

File: docs/features/proto-serializers.rst [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        queryset=Something.objects.all(),
+        pk_field=UUIDField(format="hex_verbose"),
+    )
+
+=========================================
+Converting empty string to None
+=========================================
+
+As gRPC always sends the default value for the type if not sent, some behaviors of DRF, like handling differently None value and empty string, are not working.
+You can design your own system by adding arguments to adapt the behavior, but if you have a field where an empty string means None, as for Datetime, for example, you can use code like this:
```
[the remark this answers is not in the document]
Comment (AMontagu) [reply docs/features/proto-serializers.rst:216]: @legau need you on this part. You have more knowledege about that. We should also have an example with test in fakeapp for that if not already done.

**Comment by AMontagu** [comment]
@markdoerr pass all over the TODO was meaning replacing them with actual documentation.

@markdoerr @legau finished all the remaining item. I only miss a global relecture in web format to check all the formatting and links.
If you can review the new documentation added following your feedback for grammar/spelling mistake as I didn't use gpt or any other tool and I make a lot of mistake even in french :smile: 

There is still one inner working page not done and other should follow but I think we can merge as it after your review because perfection doesn't exist and this doc is now way better than the old one.

**Comment by markdoerr** [comment]
Very good, @AMontagu,
excellent news.
I should have some time tomorrow morning to give you feedback (today is unfortunately very packed as you noticed by my delayed reply ;(

**Comment by markdoerr** [comment]
Hi @AMontagu and @legau,

here a more general remark that might be made clear in the documentation:
Since you are planning to deprecate all sync parts in favour of async, shouldn't that be really made explicit on all occasions, where you still explain sync classes / concepts ? 
If you want, I can point to the parts of the documentation, where the deprecation hint is missing.

**Comment by markdoerr** [comment]
... there is no documentation for inner-workings / Service And Servicer Proxy
are you planning to add it in this round ?

**Comment by markdoerr** [comment]
Hi @AMontagu and @legau,
I will add all my comments in my review pull request.

**Pull request merged**